### DISCUSSION TITLE
Multiple Statement SQL Procedure

### DISCUSSION THREAD

**Discussion by josephniblo** [littleK0i/SnowDDL#114] [Q&A] [answered]
Hi @littleK0i, 

I am working with SQL Procedures and would like to create one with multiple statements, because I want to execute in a single transaction.

I am getting an error like: 
> WARNING - Resolved PROCEDURE [...]: ERROR
> ...
> message   =>  Actual statement count 6 did not match the desired statement count 1

Roughly speaking, the Procedure does:
```SQL
BEGIN TRANSACTION;
CREATE OR REPLACE TEMPORARY TABLE ... ;

<SOME MORE LOGIC...>

COMMIT;
```

(with the sql written in a separate file which is referenced with `!include`)

Wondered if I have a configuration issue for my Procedure or if we are currently limited to single statement SQL Procedures in SnowDDL?

Thanks

**Comment by littleK0i** [comment, marked as answer]
@josephniblo , multi-statement procedures should definitely work. It is covered by one of [test cases](https://github.com/littleK0i/SnowDDL/blob/master/test/_config/step1/db1/sc1/procedure/pr001_pr1(number%2Cnumber%2Cnumber%2Cnumber%2Cfloat%2Cbinary%2Cbinary%2Cvarchar%2Cvarchar%2Cdate%2Ctime%2Ctime%2Ctimestamp_ltz%2Ctimestamp_ltz%2Ctimestamp_ntz%2Ctimestamp_ntz%2Ctimestamp_tz%2Ctimestamp_tz%2Cvariant%2Cobject%2Carray%2Cgeography%2Cgeometry).yaml).

Is there any chance you have an extra semi-colon in procedure body? I suspect Snowflake thinks `CREATE PROCEDURE` command was finished and some other commands were added after that. Submitting multiple commands is not allowed with basic `execute()` by default, and this is a good thing. It protects you from destroying something accidentally.

**Reply by josephniblo** [reply]
Yes whoops - was missing a `BEGIN ... END` for the Procedure... It needed to be 
```SQL
BEGIN
BEGIN TRANSACTION;
...
COMMIT;
END;
```

No SnowDDL issue.

Thanks
